### PR TITLE
[KYUUBI #2628][FOLLOWUP] Reuse the kyuubi-ctl batch commands for SubmitBatchCommand

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

For SubmitBatchCommand, reuse the CreateBatchCommand and LogBatchCommand.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/cmd/BatchCommand.scala [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+abstract class BatchCommand(cliConfig: CliConfig) extends Command(cliConfig) {
+
+  override def run(): Unit = {
+    doRunAndReturnBatchReport.foreach { batch =>
+      info(Render.renderBatchInfo(batch))
+    }
+  }
+
+  def doRunAndReturnBatchReport(): Option[Batch]
```
Comment (turboFei) [inline kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/cmd/BatchCommand.scala:32]: now ListBatchesCommand does not extend `BatchCommand`

Comment (turboFei) [reply kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/cmd/BatchCommand.scala:32]: cc @ulysses-you

Comment (turboFei) [reply kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/cmd/BatchCommand.scala:32]: ListBatchesCommand should be `BatchesCommand`

**Comment by turboFei** [comment]
cc @lightning-L 
Hope it can simplify the following optimization for SubmitBatchCommand and LogBatchCommand.

File: kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/cmd/delete/DeleteBatchCommand.scala [hunk 2]
```diff
@@ -35,16 +36,22 @@ class DeleteBatchCommand(cliConfig: CliConfig) extends Command(cliConfig) {
...
-        }
+      if (result.isSuccess) {
+        null
+      } else {
+        batchRestApi.getBatchById(batchId)
       }
     }
   }
 
+  override def run(): Unit = {
```
Comment (ulysses-you) [inline kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/cmd/delete/DeleteBatchCommand.scala:47]: can not we do this inside `doRunAndReturnBatchReport ` ?

File: kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/cmd/BatchCommand.scala [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+import org.apache.kyuubi.client.api.v1.dto.Batch
+import org.apache.kyuubi.ctl.CliConfig
+import org.apache.kyuubi.ctl.util.Render
+
+abstract class BatchCommand(cliConfig: CliConfig) extends Command(cliConfig) {
+
+  def doRunAndReturnBatchReport(): Batch
+
+  override def run(): Unit = {
```
Comment (ulysses-you) [inline kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/cmd/BatchCommand.scala:28]: shall we make it `final` ? otherwise it just add complexity ..

Comment (turboFei) [reply kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/cmd/BatchCommand.scala:28]: ok

**Comment by lightning-L** [comment]
I agree that we need to reuse the code in CreateBatchCommand and LogBatchCommand for submit batch operation.
But I am wondering if it is necessary to add a BatchCommand.

I think the main point is that we need to separate the calling api(do Run) part and the render part. Currently, those are stuck together in run() method.

For all the Command, the operation is: validate, doRun, and render.
We could separate the current run() method into doRun and render.
We could let doRun return object, and let render print the object.
It will apply to all kinds of resources. So we could do it in Command.scala instead of adding a new BatchCommand.scala?

**Comment by turboFei** [comment]
> I agree that we need to reuse the code in CreateBatchCommand and LogBatchCommand for submit batch operation. But I am wondering if it is necessary to add a BatchCommand.
> 
> I think the main point is that we need to separate the calling api(do Run) part and the render part. Currently, those are stuck together in run() method.
> 
> For all the Command, the operation is: validate, doRun, and render. We could separate the current run() method into doRun and render. We could let doRun return object, and let render print the object. It will apply to all kinds of resources. So we could do it in Command.scala instead of adding a new BatchCommand.scala?

thanks @lightning-L

**Comment by turboFei** [comment]
thanks @lightning-L 
could you have a review again?

**Review by ulysses-you — APPROVED** [review]

**Comment by turboFei** [comment]
thanks, merging to master

**Pull request closed without merging**